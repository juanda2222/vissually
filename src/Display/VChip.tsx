import React, { useContext } from 'react'
import styled  from 'styled-components';

import styles from './VChip.module.css'
import { Color2Vec } from "../Tools/ColorTools"
import { DefaultThemes, ThemeContext } from "../ThemeProvider/ThemeProvider"
import XSymbol from "../assets/x-symbol"

// style the container
const StyledContainer = styled("div")<{background_color:string, hover_background_color:string}>`
  background-color: ${props => props.background_color};
  &:hover{
    background-color: ${props => props.hover_background_color};  
  } 
  `;

    
const StyledCancelButton = styled("div") <{
  background_color: string,
  hover_background_color: string,
  text_color: string
}>`
  background-color: ${props => props.background_color};
  text-color: ${props => props.text_color};
  &:hover{
    background-color: ${props => props.hover_background_color};  
  }
`;

const VChip: React.FunctionComponent<ChipProps> = ({
  //styling properties
  className,
  style,
  theme,
  primary, secundary, dark,

  //functional properties
  label,
  onDelete,
  onClick,
}) => {

  // get the theme from the context
  const contex_theme = useContext(ThemeContext);
  const wrapped_className = [styles.container, className ? className : ""].join(" ")

  // generat the theme depending on the boolean inputs or the theme input
  var current_theme:Theme;
  if (secundary) {
    current_theme = DefaultThemes.secundary
  } else if (dark) {
    current_theme = DefaultThemes.dark
  } else if (primary) {
    current_theme = DefaultThemes.primary
  } else if (!(typeof (contex_theme) == "undefined")) {
    current_theme = contex_theme
  } else {
    current_theme = theme ? theme : DefaultThemes.primary
  }
  
  // generate the component from the style
  const rgb_list = Color2Vec(current_theme.backgroundColor1)

  return StyledContainer ? (
    <StyledContainer
      background_color={`rgb(${rgb_list[0] - 20}, ${rgb_list[1] - 20}, ${rgb_list[2] - 20})`}
      hover_background_color={`rgb(${rgb_list[0] - 10}, ${rgb_list[1] - 10}, ${rgb_list[2] - 10})`}
      className={wrapped_className}
      style={style}
      onClick={onClick}
    >
      <span>{label}</span>
      {
        onDelete &&
        <StyledCancelButton
          background_color={`rgb(${rgb_list[0] - 40}, ${rgb_list[1] - 40}, ${rgb_list[2] - 40})`}
          hover_background_color={`rgb(${rgb_list[0] - 30}, ${rgb_list[1] - 30}, ${rgb_list[2] - 30})`}
          text_color={current_theme.textColor1}
          className={styles.delete_button}
          onClick={(e) => {
            e.stopPropagation(); 
            onDelete && onDelete(e)
          }}
        >
          <XSymbol style={{
            fill: current_theme.textColor1,
            width: "0.6em", height: "0.6em",
            margin: "auto 6px"
          }} />
        </StyledCancelButton>
      }
    </StyledContainer>
  ):null
}


export default VChip