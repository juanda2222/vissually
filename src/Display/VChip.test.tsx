
import React from "react"
import renderer from 'react-test-renderer';
import VChip from './VChip'

describe('VButton', () => {

  test('Button declaration has no errors', () => {
    expect(VChip).toBeTruthy()
  })

  test("Default render. no list given", () => { 
    const Chip = renderer.create(<VChip label="snapshot"/>).toJSON();
    console.log(Chip)
    expect(Chip).toMatchSnapshot();
  })

  test("Rendering with props. Snapshot test", () => { 

    const ChipList = renderer.create(
      <div>
        <VChip label="label!"/>
        <VChip label="other one" onClick={()=>{console.log("Clicked on 'other one'")}}/>
        <VChip label="delete me" onClick={()=>{console.log("Clicked on 'delete me'")}} onDelete={()=>{console.log("deleting")}}/>
      </div>
    ).toJSON();
    console.log(ChipList)
    expect(ChipList).toMatchSnapshot();
  })


})
