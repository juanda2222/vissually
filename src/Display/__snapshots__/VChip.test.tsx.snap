// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VButton Default render. no list given 1`] = `
<div
  className="sc-AxjAm dkuJXE container "
>
  <span />
</div>
`;

exports[`VButton Rendering with props. Snapshot test 1`] = `
<div>
  <div
    className="sc-AxjAm dkuJXE container "
  >
    <span>
      label!
    </span>
  </div>
  <div
    className="sc-AxjAm dkuJXE container "
    onClick={[Function]}
  >
    <span>
      other one
    </span>
  </div>
  <div
    className="sc-AxjAm dkuJXE container "
    onClick={[Function]}
  >
    <span>
      delete me
    </span>
    <div
      className="sc-AxirZ kWaCOc delete_button"
      onClick={[Function]}
    >
      <svg
        height="24"
        style={
          Object {
            "fill": "rgb(0, 0, 0)",
            "height": "0.6em",
            "margin": "auto 6px",
            "width": "0.6em",
          }
        }
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
        />
      </svg>
    </div>
  </div>
</div>
`;
